internal/provider/provider.go

```diff
@@ -196,50 +240,147 @@ func (p *pingfederateProvider) Configure(ctx context.Context, req provider.Confi
 		}
 	}
 
-	// User must provide a username to the provider
+	// Check if the user has provided a username to the provider
 	var username string
-	if config.Username.IsUnknown() {
+	if config.Username.IsNull() {
+		username = os.Getenv("PINGFEDERATE_PROVIDER_USERNAME")
+	} else {
+		username = config.Username.ValueString()
+	}
+
+	// Check if the user has provided a password to the provider
+	var password string
+	if config.Password.IsNull() {
+		password = os.Getenv("PINGFEDERATE_PROVIDER_PASSWORD")
+	} else {
+		password = config.Password.ValueString()
+	}
+
+	// Check if the user has provided an access token to the provider
+	var accessToken string
+	if config.AccessToken.IsUnknown() {
 		// Cannot connect to PingFederate with an unknown value
 		resp.Diagnostics.AddError(
 			"Unable to connect to the PingFederate Server",
-			"Cannot use unknown value as username",
+			"Cannot use unknown value as access_token",
 		)
 	} else {
-		if config.Username.IsNull() {
-			username = os.Getenv("PINGFEDERATE_PROVIDER_USERNAME")
+		if config.AccessToken.IsNull() {
+			accessToken = os.Getenv("PINGFEDERATE_PROVIDER_ACCESS_TOKEN")
 		} else {
-			username = config.Username.ValueString()
+			accessToken = config.AccessToken.ValueString()
 		}
-		if username == "" {
-			resp.Diagnostics.AddError(
-				"Unable to find username",
-				"username cannot be an empty string. Either set it in the configuration or use the PINGFEDERATE_PROVIDER_USERNAME environment variable.",
-			)
+	}
+
+	// Check if the user has provided an OAuth configuration to the provider
+	var hasOauthConfig bool = false
+	var clientId string
+	var clientSecret string
+	var scopes []string
+	var tokenUrl string
+	if username == "" && password == "" && accessToken == "" {
+		configClientId, ok := config.OAuth.Attributes()["client_id"].(types.String)
```

Comment:
What happens here if the user doesn't set the oauth block at all (if they haven't set any authentication attributes)? It seems like this could end up trying to cast a nil and panicking.